include/modem/lte_lc.h

```diff
@@ -672,186 +883,240 @@ struct lte_lc_conn_eval_params {
+ * types up to @ref LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_COMPLETE, a list of neighbor cell
+ * measurement results related to the current_cell (@ref lte_lc_cells_info).
  */
 enum lte_lc_neighbor_search_type {
-	/** The modem searches the network it is registered to (RPLMN) based on
-	 *  previous cell history.
-	 *  For modem firmware versions < 1.3.1, this is the only valid option.
+	/**
+	 * The modem searches the network it is registered to (RPLMN) based on previous cell
+	 * history.
 	 */
 	LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT = 1,
 
-	/** The modem starts with the same search method as the default type.
-	 *  If needed, it continues to search by measuring the radio conditions
-	 *  and makes assumptions on where networks might be deployed, i.e. a light search.
-	 *  The search is limited to bands that are valid for the area of the current
-	 *  ITU-T region. If RPLMN is not found based on previous cell history, the
-	 *  modem accepts any found PLMN.
+	/**
+	 * The modem starts with the same search method as the default type. If needed, it
+	 * continues to search by measuring the radio conditions and makes assumptions on where
+	 * networks might be deployed, in other words, a light search. The search is limited to
+	 * bands that are valid for the area of the current ITU-T region. If RPLMN is not found
+	 * based on previous cell history, the modem accepts any found PLMN.
+	 *
+	 * @note This search type is only supported by modem firmware versions >= v1.3.1.
 	 */
 	LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_LIGHT = 2,
 
-	/** The modem follows the same procedure as for LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_LIGHT,
-	 *  but will continue to perform a complete search instead of a light search,
-	 *  and the search is performed for all supported bands.
+	/**
+	 * The modem follows the same procedure as for
+	 * @ref LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_LIGHT, but will continue to perform a complete
+	 * search instead of a light search, and the search is performed for all supported bands.
+	 *
+	 * @note This search type is only supported by modem firmware versions >= v1.3.1.
 	 */
 	LTE_LC_NEIGHBOR_SEARCH_TYPE_EXTENDED_COMPLETE = 3,
 
-	/** GCI search, option 1. Modem searches EARFCNs based on previous cell history. */
+	/**
+	 * GCI search, option 1. Modem searches EARFCNs based on previous cell history.
+	 *
+	 * @note This search type is only supported by modem firmware versions >= v1.3.4.
+	 */
 	LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_DEFAULT = 4,
 
-	/** GCI search, option 2. Modem starts with the same search method as in
-	 *  LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_DEFAULT. If less than gci_count cells were found,
-	 *  the modem performs a light search on bands that are valid for the area of
-	 *  the current ITU-T region.
+	/**
+	 * GCI search, option 2. Modem starts with the same search method as in
+	 * @ref LTE_LC_NEIGHBOR_SEARCH_TYPE_GCI_DEFAULT. If less than gci_count cells were found,
```

Comment:
If we wanted to mention that, it could be for the `gci_count` parameter in `struct lte_lc_ncellmeas_params`. The range starts from 2, which means the current cell and one surrounding cell, I guess?

Reply:
Maybe it would be good to mention that in the `gci_count` parameter description.
This is just how it works based on my experience as I've never seen as many surrounding cells as requested.

Reply:
How about this for `gci_count`?
```
	 * Current cell is counted as one cell, so the maximum number of surrounding cells searched
	 * is `gci_count - 1`. Mandatory with the GCI search types, ignored with other search types.
```

Reply:
As discussed, this could be removed to cover the case when there is no current cell if the search works then:
` so the maximum number of surrounding cells searched is 'gci_count - 1'`